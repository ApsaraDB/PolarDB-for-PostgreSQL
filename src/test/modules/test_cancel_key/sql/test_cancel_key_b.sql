-- This is a proxy process
select pg_sleep(0.5); -- 0.5s
SELECT test_cancel_key(true);

select pg_sleep(0.5); -- 1.0s
select pid from pg_stat_activity where pid = 0; -- None (b is unset and will return real pid)
select pid from pg_stat_activity where pid > 10000000 order by pid; -- 10000009

select pg_sleep(0.5); -- 1.5s
set polar_virtual_pid = 10000001;
select pid from pg_stat_activity where pid = 0; -- None
select pid from pg_stat_activity where pid > 10000000 order by pid; -- 10000001 and 10000009

select pg_sleep(2); -- 3.5 and will be canceled
