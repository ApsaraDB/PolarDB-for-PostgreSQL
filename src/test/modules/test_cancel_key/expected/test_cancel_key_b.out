-- This is a proxy process
select pg_sleep(0.5); -- 0.5s
 pg_sleep 
----------
 
(1 row)

SELECT test_cancel_key(true);
 test_cancel_key 
-----------------
 
(1 row)

select pg_sleep(0.5); -- 1.0s
 pg_sleep 
----------
 
(1 row)

select pid from pg_stat_activity where pid = 0; -- None (b is unset and will return real pid)
 pid 
-----
(0 rows)

select pid from pg_stat_activity where pid > 10000000 order by pid; -- 10000009
   pid    
----------
 10000009
(1 row)

select pg_sleep(0.5); -- 1.5s
 pg_sleep 
----------
 
(1 row)

set polar_virtual_pid = 10000001;
select pid from pg_stat_activity where pid = 0; -- None
 pid 
-----
(0 rows)

select pid from pg_stat_activity where pid > 10000000 order by pid; -- 10000001 and 10000009
   pid    
----------
 10000001
 10000009
(2 rows)

select pg_sleep(2); -- 3.5 and will be canceled
ERROR:  canceling statement due to user request
